packages/react-dev-utils/README.md

```diff
@@ -0,0 +1,180 @@
+
+// Webpack config
+module.exports = {
+  // ...
+  plugins: [
+    // ...
+    // If you require a missing module and then `npm install` it, you still have
+    // to restart the development server for Webpack to discover it. This plugin
+    // makes the discovery automatic so you don't have to restart.
+    // See https://github.com/facebookincubator/create-react-app/issues/186
+    new WatchMissingNodeModulesPlugin(path.resolve('node_modules'))
+  ],
+  // ...
+}
+```
+
+#### `checkRequiredFiles(files: Array<string>): boolean`
+
+Makes sure that all passed files exist.  
+Filenames are expected to be absolute.  
+If a file is not found, prints a warning message and returns `false`.
+
+```js
+var path = require('path');
+var checkRequiredFiles = require('react-dev-utils/checkRequiredFiles');
+
+if (!checkRequiredFiles([
+  path.resolve('public/index.html'),
+  path.resolve('src/index.js')
+])) {
+  process.exit(1);
+}
+```
+
+#### `clearConsole(): void`
+
+Clears the console, hopefully in a cross-platform way.
+
+```js
+var clearConsole = require('react-dev-utils/clearConsole');
+
+clearConsole();
+console.log('Just cleared the screen!');
+```
+
+#### `formatWebpackMessages(stats: WebpackStats): {errors: Array<string>, warnings: Array<string>}`
+
+Extracts and prettifies warning and error messages from webpack [stats](https://github.com/webpack/docs/wiki/node.js-api#stats) object.
+
+```js
+var webpack = require('webpack');
+var config = require('../config/webpack.config.dev');
+
+var compiler = webpack(config);
+
+compiler.plugin('invalid', function() {
+  console.log('Compiling...');
+});
+
+compiler.plugin('done', function(stats) {
+  var messages = formatWebpackMessages(stats);
+  if (!messages.errors.length && !messages.warnings.length) {
+    console.log('Compiled successfully!');
+  }
+  if (messages.errors.length) {
+    console.log('Failed to compile.');
+    messages.errors.forEach(console.log);
+    return;
+  }
+  if (messages.warnings.length) {
+    console.log('Compiled with warnings.');
+    messages.warnings.forEach(console.log);
+  }
+});
+```
+
+#### `openBrowser(url: string): boolean`
+
+Attempts to open the browser with a given URL.  
+On Mac OS X, attempts to reuse an existing Chrome tab via AppleScript.  
+Otherwise, falls back to [opn](https://github.com/sindresorhus/opn) behavior.
+
+
+```js
+var path = require('path');
+var openBrowser = require('react-dev-utils/openBrowser');
+
+if (openBrowser('http://localhost:3000')) {
+  console.log('The browser tab has been opened!');
+}
+```
+
+#### `prompt`
+
+This function displays a console prompt to the user.
+
+By convention, "no" should be the conservative choice.  
+If you mistype the answer, we'll always take it as a "no".  
+You can control the behavior on `<Enter>` with `isYesDefault`.
+
+```js
+var prompt = require('react-dev-utils/prompt');
+prompt(
+  'Are you sure you want to eat all the candy?',
+  false
```

Comment:
I always put a comment for boolean variables with what they mean, otherwise it's pretty confusing :)

```js
/* isYesDefault*/ false
```